test/framework/cluster.go

```diff
@@ -220,19 +220,43 @@ func (e *ClusterE2ETest) PowerOffHardware(opts ...CommandOpt) {
 	ctx := context.Background()
 
 	for _, h := range e.TestHardware {
-		bmcInfo := pbnj.BmcSecretConfig{
-			Host:     h.BmcIpAddress,
-			Username: h.BmcUsername,
-			Password: h.BmcPassword,
-			Vendor:   h.BmcVendor,
-		}
+		bmcInfo := api.NewBmcSecretConfig(h)
 		err := pbnjClient.PowerOff(ctx, bmcInfo)
 		if err != nil {
 			e.T.Fatalf("failed to power off hardware: %v", err)
 		}
 	}
 }
 
+func (e *ClusterE2ETest) ValidateHardwareDecommissioned() {
+	pbnjEndpoint := os.Getenv(tinkerbellPBnJGRPCAuthEnvVar)
+	pbnjClient, err := pbnj.NewPBNJClient(pbnjEndpoint)
+	if err != nil {
+		e.T.Fatalf("failed to create pbnj client: %v", err)
+	}
+
+	ctx := context.Background()
+
+	var failedToDecomm []*api.Hardware
+	for _, h := range e.TestHardware {
```

Comment:
How doe this trigger? Should we be instrumenting retry logic? Dependent on timing this could be extremely brittle.